bdd-generator/src/main/java/com/epam/test_generator/services/StepService.java

```diff
@@ -108,20 +120,26 @@ public StepDTO addStepToCase(Long projectId, Long suitId, Long caseId,
      * @param stepId id of step which to update
      */
     public StepDTO updateStep(Long projectId, Long suitId, Long caseId, Long stepId,
-                           StepUpdateDTO stepUpdateDTO) {
+                              StepUpdateDTO stepUpdateDTO) {
         Suit suit = suitService.getSuit(projectId, suitId);
         Case caze = caseService.getCase(projectId, suitId, caseId);
 
-        Step step = stepDAO.findById(stepId).orElseThrow(NotFoundException::new);
-        throwExceptionIfStepIsNotInCase(caze, step);
-
-        step = stepTransformer.updateFromDto(stepUpdateDTO, step);
-        stepDAO.save(step);
-
+        Step oldStep = stepDAO.findById(stepId).orElseThrow(NotFoundException::new);
+        throwExceptionIfStepIsNotInCase(caze, oldStep);
+        StepSuggestion oldStepSuggestion = stepSuggestionService.getStepSuggestion(projectId, oldStep);
+        Step newStep = stepTransformer.updateFromDto(stepUpdateDTO, oldStep);
+        StepSuggestion newStepSuggestion = stepSuggestionService.getStepSuggestion(projectId, newStep);
+        stepDAO.save(newStep);
+        if(!Objects.equals(oldStepSuggestion, newStepSuggestion)) {
+            oldStepSuggestion.getSteps().remove(oldStep);
```

Comment:
In this case, you abuse getSteps() excess functionality, so if you want to make a refactoring in the future, you'll have to reimplement full functionality of getSteps() return value. I suggest using 'remove' and 'add' methods of StepSuggestion: 
```
newStepSuggestion.remove(oldStep);
newStepSuggestion.remove(newStep);
```

Reply:
Fixed